src/core/mac/mac.cpp

```diff
@@ -1571,7 +1635,9 @@ void Mac::ReceiveDoneTask(Frame *aFrame, otError aError)
         if (!mRxOnWhenIdle && dstaddr.mLength != 0)
         {
             mReceiveTimer.Stop();
-            otPlatRadioSleep(GetInstance());
+            mWaitingForData = false;
+            SetFramePending(aFrame->GetFramePending());
+            mScheduleNextTrasmission = true;
```

Comment:
If I am not mistaken, we may be able to convert `mScheduleNextTrasmission` into a local variable  in this method instead of it being a member/instance variable. I think it's used only in this method.